### PR TITLE
refactor: move system tests into `tests/system` directory

### PR DESCRIPTION
**Pull request by tswast**
This aligns more closely with the templates from synthtool.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Towards #366

### CODE REVIEW

**Comment by tswast** [comment]
prerelease-deps-3.8 test failures are both in the _unit_ tests. https://source.cloud.google.com/results/invocations/aed3c05e-366a-404e-b90a-ecb134327e41/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fpython-bigquery%2Fpresubmit%2Fprerelease-deps-3.8/log

Neither seem particularly related to pyarrow (failure due to bad nightly) or to this change.

Perhaps there's some state that's being held onto when the unit tests are run in the same `pytest` run as the system tests?

File: tests/system/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2020 Google LLC
```
Comment (plamut) [inline tests/system/__init__.py:1]: (nit) IIRC we need to use the file creation year which is 2021? :)

```suggestion
# Copyright 2021 Google LLC
```

File: tests/system/test_magics.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+import pytest
+import psutil
+
+
+IPython = pytest.importorskip("IPython")
+io = pytest.importorskip("IPython.utils.io")
+pandas = pytest.importorskip("pandas")
+tools = pytest.importorskip("IPython.testing.tools")
+interactiveshell = pytest.importorskip("IPython.terminal.interactiveshell")
```
Comment (plamut) [inline tests/system/test_magics.py:27]: Nice, this is IMO more clean that those `skipIf` decorators. :+1:

File: tests/system/test_client.py [hunk 3]
```diff
@@ -76,8 +77,8 @@
 from test_utils.system import unique_resource_id
 
 
-JOB_TIMEOUT = 120  # 2 minutes
-WHERE = os.path.abspath(os.path.dirname(__file__))
+JOB_TIMEOUT = 120  # 2 minute
+DATA_PATH = pathlib.Path(__file__).parent.parent / "data"
```
Comment (plamut) [inline tests/system/test_client.py:81]: Love these small improvements and utilizing the goodies Python 3 offers!

**Review by plamut — APPROVED** [review]
Apart from a license header nit, it looks good to me. :+1:

**Pull request merged**